tests/framework/request.go

```diff
@@ -16,14 +17,27 @@ import (
 func Get(url, address string, timeout time.Duration) (int, string, error) {
 	dialer := &net.Dialer{}
 
-	http.DefaultTransport.(*http.Transport).DialContext = func(
-		ctx context.Context,
-		network,
-		addr string,
-	) (net.Conn, error) {
-		split := strings.Split(addr, ":")
-		port := split[len(split)-1]
-		return dialer.DialContext(ctx, network, fmt.Sprintf("%s:%s", address, port))
+	transport := &http.Transport{
+		DialContext: func(
+			ctx context.Context,
+			network,
+			addr string,
+		) (net.Conn, error) {
+			split := strings.Split(addr, ":")
+			port := split[len(split)-1]
+			return dialer.DialContext(ctx, network, fmt.Sprintf("%s:%s", address, port))
+		},
+	}
+
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   timeout,
+	}
+
+	if strings.HasPrefix(url, "https") {
+		t := client.Transport.(*http.Transport).Clone()
+		t.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		client.Transport = t
```

Comment:
since the default client available via `http` is shared, not modifying it for security settings looks reasonable to me

we do have it in the main branch now though -- https://github.com/nginxinc/nginx-gateway-fabric/blob/4d5c663c6469ea6a5507492d29cd561ce2432d82/tests/framework/request.go#L39-L45